clientUtils/patchHelper.ts

```diff
@@ -0,0 +1,106 @@
+import jsonpointer from "json8-pointer"
+import { VariableAnnotationPatch } from "./AdminSessionTypes"
+import { isArray, isPlainObjectWithGuard } from "./Util"
+export function setValueRecursive(
+    json: any,
+    pointer: string[],
+    newValue: any
+): any {
+    if (pointer.length == 0) {
+        return newValue
+    } else {
+        const currentPart = pointer[0]
+        const currentPartAsNumber = Number.parseInt(currentPart)
+        if (Number.isNaN(currentPartAsNumber)) {
+            let newObject: any = {}
+            if (json !== undefined) newObject = { ...json }
+            const currentValue = newObject.hasOwnProperty(currentPart)
+                ? newObject[currentPart]
+                : undefined
+            const updatedValue = setValueRecursive(
+                currentValue,
+                pointer.slice(1),
+                newValue
+            )
+            if (updatedValue === null) delete newObject[currentPart]
```

Comment:
As I understand, setValue with `null` value means delete the property?

Does this come from an external spec, or is it our interpretation?

I'm wondering because `null` is a valid JSON value, and in some exceptional circumstances we may want to use it? On the other hand, `undefined` isn't a valid JSON value, so it _seems_ better to use that to delete properties.